src/libtiled/tilelayer.h

```diff
@@ -331,19 +386,16 @@ class TILEDSHARED_EXPORT TileLayer : public Layer
 
     TileLayer *clone() const override;
 
-    // Enable easy iteration over cells with range-based for
-    QVector<Cell>::iterator begin() { return mGrid.begin(); }
-    QVector<Cell>::iterator end() { return mGrid.end(); }
-    QVector<Cell>::const_iterator begin() const { return mGrid.begin(); }
-    QVector<Cell>::const_iterator end() const { return mGrid.end(); }
+    void copyGrid(const QVector<Cell> &newGrid);
 
 protected:
     TileLayer *initializeClone(TileLayer *clone) const;
 
 private:
     int mWidth;
     int mHeight;
-    QVector<Cell> mGrid;
+    Cell mEmptyCell;
+    QMap< QPair<int, int>, Block* > mMap;
```

Comment:
I don't see how it matters whether `QMap` or `QHash` is used. Both can be iterated just fine.